delivery/jwt.py

```diff
@@ -0,0 +1,192 @@
+    RS256 = 'RS256'
+    HS256 = 'HS256'
+
+
+@dataclasses.dataclass(frozen=True)
+class JSONWebKey:
+    kty: KeyType
+    use: Use
+    alg: Algorithm
+    kid: str
+
+    @property
+    def key(self) -> bytes:
+        # has to be defined by derived classes
+        pass
+
+    @staticmethod
+    def from_dict(data: dict) -> 'JSONWebKey':
+        algorithm = Algorithm(data.get('alg'))
+
+        class_by_algorithm = {
+            Algorithm.RS256: RSAPublicKey,
+            Algorithm.HS256: SymmetricKey,
+        }
+
+        return dacite.from_dict(
+            data_class=class_by_algorithm.get(algorithm),
+            data=data,
+            config=dacite.Config(
+                cast=[enum.Enum],
+            ),
+        )
+
+    @staticmethod
+    def from_signing_cfg(signing_cfg: model.delivery.SigningCfg) -> 'JSONWebKey':
+        algorithm = Algorithm(signing_cfg.algorithm().upper())
+        use = Use.SIGNATURE
+        kid = signing_cfg.id()
+
+        if algorithm == Algorithm.RS256:
+            public_key = Crypto.PublicKey.RSA.import_key(signing_cfg.public_key())
+
+            return RSAPublicKey(
+                use=use,
+                kid=kid,
+                n=encodeBase64urlUInt(public_key.n),
+                e=encodeBase64urlUInt(public_key.e),
+            )
+        elif algorithm == Algorithm.HS256:
+            return SymmetricKey(
+                use=use,
+                kid=kid,
+                k=encodeBase64url(signing_cfg.secret().encode('utf-8')),
+            )
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class RSAPublicKey(JSONWebKey):
+    n: str # modulus (Base64urlUInt encoded)
+    e: str # exponent (Base64urlUInt encoded)
+    kty: KeyType = KeyType.RSA
+    alg: Algorithm = Algorithm.RS256
+
+    @property
+    def key(self) -> bytes:
+        return Crypto.PublicKey.RSA.construct(
+            rsa_components=(
+                decodeBase64urlUInt(self.n),
+                decodeBase64urlUInt(self.e),
+            ),
+        ).export_key(format='PEM')
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class SymmetricKey(JSONWebKey):
+    k: str # key value (base64url encoded)
+    kty: KeyType = KeyType.OCTET_SEQUENCE
+    alg: Algorithm = Algorithm.HS256
+
+    @property
+    def key(self) -> bytes:
+        return decodeBase64url(self.k)
+
+
+def encodeBase64url(b: bytes) -> str:
+    url_encoding = base64.urlsafe_b64encode(b)
+
+    return url_encoding.rstrip(b'=').decode('utf-8')
+
+
+def decodeBase64url(s: str) -> bytes:
+    return base64.urlsafe_b64decode(s + '==')
+
+
+def encodeBase64urlUInt(number: int) -> str:
+    b = Crypto.Util.number.long_to_bytes(number)
+
+    return encodeBase64url(b)
+
+
+def decodeBase64urlUInt(s: str) -> int:
+    b = decodeBase64url(s)
+
+    return Crypto.Util.number.bytes_to_long(b)
+
+
+@functools.cache
+def decode_jwt(
+    token: str,
+    verify_signature: bool=True,
+    json_web_key: JSONWebKey | None=None,
+    **kwargs,
```

Comment:
add docstr stating this function wraps jwt.decode (esp. so callers know about the allowed kwargs)